packit_service/worker/build/copr_build.py

```diff
@@ -167,71 +166,39 @@ def run_copr_build(self) -> HandlerResults:
-        ex: Optional[Exception] = None  # shut up pycharm
-        extra_logs: str = ""
+    def run_build(
+        self, target: Optional[str] = None
+    ) -> Tuple[Optional[int], Optional[str]]:
+        """
+        Trigger the build and return id and web_url
+        :param target: str, run for all if not set
+        :return: task_id, task_url
+        """
 
-        try:
-            c.copr_build_id, c.copr_web_url = self.api.run_copr_build(
-                project=self.job_project,
-                chroots=self.build_chroots,
-                owner=self.job_owner,
+        owner = self.job_owner or self.api.copr_helper.configured_owner
+        if not owner:
+            raise PackitCoprException(
+                f"Copr owner not set. Use Copr config file or `--owner` when calling packit CLI."
             )
-        except SandcastleTimeoutReached as e:
-            ex = e
-            extra_logs = f"\nYou have reached 10-minute timeout while creating SRPM.\n"
-        except ApiException as e:
-            ex = e
-            # this is an internal error: let's not expose anything to public
-            extra_logs = (
-                "\nThere was a problem in the environment the packit-service is running in.\n"
-                "Please hang tight, the help is coming."
-            )
-        except Exception as e:
-            ex = e  # shut up mypy
-
-        # collect the logs now
-        packit_logger.removeHandler(handler)
-        stream.seek(0)
-        c.srpm_logs = stream.read()
-
-        if ex:
-            logger.info(f"Exception while running a copr build: {ex}")
-            logger.debug(f"{ex!r}")
-
-            c.srpm_failed = True
 
-            # when do we NOT want to send stuff to sentry?
-            sentry_integration.send_to_sentry(ex)
-
-            # this needs to be done AFTER we gather logs
-            # so that extra logs are after actual logs
-            c.srpm_logs += extra_logs
-            if hasattr(ex, "output"):
-                output = getattr(ex, "output", "")  # mypy
-                c.srpm_logs += f"\nOutput of the command in the sandbox:\n{output}\n"
-
-            c.srpm_logs += (
-                f"\nMessage: {ex}\nException: {ex!r}\n{self.msg_retrigger}"
-                "\nPlease join the freenode IRC channel #packit for the latest info.\n"
-            )
+        self.api.copr_helper.create_copr_project_if_not_exists(
+            project=self.job_project,
+            chroots=self.build_chroots,
+            owner=owner,
+            description=None,
+            instructions=None,
+        )
+        logger.debug(
+            f"owner={owner}, project={self.job_project}, path={self.srpm_path}"
+        )
 
-        return c
+        build = self.api.copr_helper.copr_client.build_proxy.create_from_file(
+            ownername=owner, projectname=self.job_project, path=self.srpm_path
+        )
+        return build.id, self.api.copr_helper.copr_web_build_url(build)
```

Comment:
I see this method is mostly a copy of [PackitAPI.run_copr_build()](https://github.com/packit-service/packit/blob/master/packit/api.py#L639). Would it make sense to refactor it in the future so that we don't have the duplicated code here and in PackitAPI?

Reply:
Yes, the reason is to exclude the srpm creation.

Reply:
OK, created 'refactoring' packit issue: https://github.com/packit-service/packit/issues/831